server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessor.java

```diff
@@ -213,8 +213,6 @@ public void insert(InsertRowPlan insertRowPlan) throws WriteProcessException {
       checkMemCostAndAddToTspInfo(insertRowPlan);
     }
 
-    workMemTable.insert(insertRowPlan);
-
     if (IoTDBDescriptor.getInstance().getConfig().isEnableWal()) {
       try {
         getLogNode().write(insertRowPlan);
```

Comment:
Move the wal to the front of the memCost check?

Reply:
fixed